di-ipv-cimit-stub/lambdas/get-contra-indicators/src/main/java/uk/gov/di/ipv/core/getcontraindicators/GetContraIndicatorsHandler.java

```diff
@@ -1,22 +1,41 @@
 package uk.gov.di.ipv.core.getcontraindicators;
 
 import com.amazonaws.services.lambda.runtime.Context;
-import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.StringMapMessage;
 import uk.gov.di.ipv.core.getcontraindicators.domain.GetCiRequest;
 import uk.gov.di.ipv.core.getcontraindicators.domain.GetCiResponse;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.util.Collections;
 
-public class GetContraIndicatorsHandler implements RequestHandler<GetCiRequest, GetCiResponse> {
+public class GetContraIndicatorsHandler implements RequestStreamHandler {
 
     private static final Logger LOGGER = LogManager.getLogger();
+    private static final ObjectMapper mapper = new ObjectMapper();
 
     @Override
-    public GetCiResponse handleRequest(GetCiRequest event, Context context) {
-        LOGGER.info(new StringMapMessage().with("EVENT TYPE:", event.getClass().toString()));
-        return new GetCiResponse(Collections.emptyList());
+    public void handleRequest(InputStream input, OutputStream output, Context context)
+            throws IOException {
+        LOGGER.info(new StringMapMessage().with("Function invoked:", "GetContraIndicators"));
+
+        GetCiResponse response;
+        try {
+            mapper.readValue(input, GetCiRequest.class);
+        } catch (Exception ex) {
+            LOGGER.error(
+                    new StringMapMessage().with("Unable to parse input request", ex.getMessage()));
+            throw ex;
```

Comment:
I think, we can turn back a custom exception or a specific failure response. It's better what we can check to SPOT team response ant hen same response like them.   generateJWT(), getPrivateKey() can catch exception details and throw exception to the our custom exception with error details.

Reply:
can this be done later once checked with spot team.

Reply:
Yes, we can handle it after the spot team is done. If you want, we can create a ticket about this.